doc/service_config.md

```diff
@@ -13,12 +13,20 @@ The service config is a JSON string of the following form:
 ```
 {
   // Load balancing policy name.
-  // Supported values are 'round_robin' and 'grpclb'.
-  // Optional; if unset, the default behavior is pick the first available
-  // backend.
-  // Note that if the resolver returns only balancer addresses and no
-  // backend addresses, gRPC will always use the 'grpclb' policy,
-  // regardless of what this field is set to.
+  // Currently, the only selectable client-side policy provided with gRPC
+  // is 'round_robin', but third parties may add their own policies.
+  // This field is optional; if unset, the default behavior is to pick
+  // the first available backend.
+  // If the policy name is set via the client API, that value overrides
+  // the value specified here.
+  //
+  // Note that if the resolver returns at least one balancer address (as
+  // opposed to backend addresses), gRPC will use grpclb (see
+  // https://github.com/grpc/grpc/blob/master/doc/load-balancing.md),
+  // regardless of what LB policy is requested either here or via the
+  // client API.  However, if the resolver returns backend addresses as
+  // well as balancer addresses, the client may fall back to the requested
+  // policy if it is unable to reach any of the grpclb load balancers.
```

Comment:
> Fair enough. Can we name it "default" or "unspecified", like we did internally? Not giving it a name just makes it harder to write and comprehend the spec because you can't concisely refer to it.

Giving the pick_first policy a name here would still allow people to set it explicitly, which is exactly what Abhishek wanted to avoid.  The only reason we have "unspecified" internally is that it's a best practice to use that name for the 0-value for a proto3 enum field, since the proto3 API (at least in C++) doesn't allow you to differentiate between the field not being set and it being explicitly set to its default value.

> Having only backend addresses also fall into the category of not having any reachable balancer address. If we always treat grpclb as the primary policy, the others does sound like a fallback.

The term "fallback" implies that we're doing something unusual because our primary mechanism did not work for some reason.  However, in cases where grpclb is intentionally not being used, I think that's misleading, because we never tried to use grpclb in the first place.

I don't think that using the term "fallback" in this field name actually makes it any easier for people to understand how this works.